### PR TITLE
Csr feat/update doc design diag

### PR DESCRIPTION
**Pull request by aggarw13**
*Issue #, if available:*

*Description of changes:*
* Update the existing design diagram for Provisioning Library to reflect recent library changes. Rename the diagram to mention "_detailed_design_" due to its high granularity of information
* Also, add a _high-level_ design diagram for the library that act as a library architecture reference (without overwhelming the reader with implementation details)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/plantuml/provisioning_operation_highlevel_design.pu [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+provisioning --> app: Return BAD_PARAMETER for any invalid input argument
+
+provisioning -> mqtt: Subscribe to 'accepted' and 'rejected' response topics\nof operation
+
+mqtt -> aws: Transmit MQTT SUBSCRIBE packets to Server
+mqtt -> provisioning: Return SUCCESS or error
+provisioning --> app: Return MQTT_ERROR if error in subscribing
+
+provisioning -> provisioning: Serialize PUBLISH payload
+provisioning -> provisioning: Return error if serialization failed.
```
Comment (leegeth) [inline doc/plantuml/provisioning_operation_highlevel_design.pu]: does this error need to be passed to the Application?

Comment (aggarw13) [reply doc/plantuml/provisioning_operation_highlevel_design.pu]: Updated the comment to be more meaningful about the error values that are passed for serialization failure back to the application.
Also, fixed the direction of the arrow to point from `provisioning` -> `app`

File: doc/plantuml/provisioning_operation_detailed_design.pu [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+parser -> callback: Return SUCCESS
+end alt
+callback -> provisioning: Notify pending API calling context about server response reception
+note left of callback: Notification done on the shared <b>semaphore resource</b>.
+alt Corner Case: Server response received after Provisionin library cleanup
+callback -> callback: Destroy semaphore resource
+note right of callback: Tackles <b>RACE CONDITION</b> between receiving\nduplicate server responses (MQTT QoS 1)\nand cleaning library after API returns.
+end alt
+deactivate callback
+activate app
```
Comment (leegeth) [inline doc/plantuml/provisioning_operation_detailed_design.pu]: This seems to be not working in the diagram generated. It is better to have the task context shown

Comment (aggarw13) [reply doc/plantuml/provisioning_operation_detailed_design.pu]: Good observation. Made minor section movements to fix the rendering issue

**Pull request merged**